internal/planner/max.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package planner
+
+import (
+	"math"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/core"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+)
+
+type maxNode struct {
+	documentIterator
+	docMapper
+
+	p    *Planner
+	plan planNode
+
+	isFloat           bool
+	virtualFieldIndex int
+	aggregateMapping  []mapper.AggregateTarget
+
+	execInfo maxExecInfo
+}
+
+type maxExecInfo struct {
+	// Total number of times maxNode was executed.
+	iterations uint64
+}
+
+func (p *Planner) Max(
+	field *mapper.Aggregate,
+	parent *mapper.Select,
+) (*maxNode, error) {
+	isFloat := false
+	for _, target := range field.AggregateTargets {
+		isTargetFloat, err := p.isValueFloat(parent, &target)
+		if err != nil {
+			return nil, err
+		}
+		// If one source property is a float, the result will be a float - no need to check the rest
+		if isTargetFloat {
+			isFloat = true
+			break
+		}
+	}
+
+	return &maxNode{
+		p:                 p,
+		isFloat:           isFloat,
+		aggregateMapping:  field.AggregateTargets,
+		virtualFieldIndex: field.Index,
+		docMapper:         docMapper{field.DocumentMapping},
+	}, nil
+}
+
+func (n *maxNode) Kind() string           { return "maxNode" }
+func (n *maxNode) Init() error            { return n.plan.Init() }
+func (n *maxNode) Start() error           { return n.plan.Start() }
+func (n *maxNode) Spans(spans core.Spans) { n.plan.Spans(spans) }
```

Comment:
suggestion: Would really like this to be hit with the tests

Reply:
should be covered now. I noticed that every other aggregate is also missing this coverage